src/batched/KokkosBatched_InverseLU_Serial_Impl.hpp

```diff
@@ -0,0 +1,196 @@
+#ifndef __KOKKOSBATCHED_INVERSELU_SERIAL_IMPL_HPP__
+#define __KOKKOSBATCHED_INVERSELU_SERIAL_IMPL_HPP__
+
+
+/// \author Vinh Dang (vqdang@sandia.gov)
+
+#include "KokkosBatched_Util.hpp"
+#include "KokkosBatched_Trsm_Decl.hpp"
+#include "KokkosBatched_Trsm_Serial_Impl.hpp"
+
+
+namespace KokkosBatched {
+  namespace Experimental {
+    ///
+    /// Serial Impl
+    /// =========
+
+    ///
+    /// InverseLU no piv
+    ///
+
+#if                                                     \
+  defined(__KOKKOSBATCHED_INTEL_MKL__) &&               \
+  defined(__KOKKOSBATCHED_INTEL_MKL_BATCHED__) &&       \
+  defined(__KOKKOSBATCHED_INTEL_MKL_COMPACT_BATCHED__)
+    template<>
+    template<typename AViewType,
+             typename WViewType>
+    KOKKOS_INLINE_FUNCTION
+    int
+    SerialInverseLU<Algo::InverseLU::CompactMKL>::
+    invoke(const AViewType &A,
+           const WViewType &W) {
+      typedef typename AViewType::value_type vector_type;
+      //typedef typename vector_type::value_type value_type;
+
+      const int
+        m = A.extent(0),
+        n = A.extent(1);
+
+      static_assert(is_vector<vector_type>::value, "value type is not vector type");      
+      static_assert(vector_type::vector_length == 4 || vector_type::vector_length == 8, 
+                    "AVX, AVX2 and AVX512 is supported");
+      assert(m==n);
+
+      const MKL_COMPACT_PACK format = vector_type::vector_length == 8 ?  MKL_COMPACT_AVX512 : MKL_COMPACT_AVX;
+
+      int r_val = 0;
+      if (A.stride_0() == 1) {
+        //mkl_dgetrfnp_compact(MKL_COL_MAJOR, 
+        //                     m, n, 
+        //                     (double*)A.data(), A.stride_1(), 
+        //                     (MKL_INT*)&r_val, format, (MKL_INT)vector_type::vector_length);
+
+        //void mkl_dgetrfnp_compact (MKL_LAYOUT layout, MKL_INT m, MKL_INT n, double * ap, MKL_INT ldap, 
+        //                           MKL_INT * info, MKL_COMPACT_PACK format, MKL_INT nm);
+
+        //void mkl_dgetrinp_compact (MKL_LAYOUT layout, MKL_INT n, double * ap, MKL_INT ldap, double * work, 
+        //                           MKL_INT lwork, MKL_INT * info, MKL_COMPACT_PACK format, MKL_INT nm);
+        mkl_dgetrinp_compact (MKL_COL_MAJOR, n, 
+                              (double*)A.data(), A.stride_1(), 
+                              (double*)W.data(), n*n, 
+                              (MKL_INT*)&r_val, format, (MKL_INT)vector_type::vector_length);
+
+      } else if (A.stride_1() == 1) {
+        //mkl_dgetrfnp_compact(MKL_ROW_MAJOR, 
+        //                     m, n, 
+        //                     (double*)A.data(), A.stride_0(), 
+        //                     (MKL_INT*)&r_val, format, (MKL_INT)vector_type::vector_length);
```

Comment:
@vqd8a did you test this intel functions ? The workspace definition might be different from what we do here. You should check the workspace size (currently n*n) needs to account vector_length (could also be n*n*vector_length).

Reply:
Also in this interface, you did not check any view rank nor span (they are necessary). Also, mkl interface only works with continuous 1D view. For better unification of interface, it might be better to remove 2D workview interface.

Reply:
The workspace query returns the workspace size is n * vector_length. Just to be safe, use n * n * vector_length in the code.